moon-store-mac-os/moon-store-mac-os/Helpers/MSError+Helper.swift

```diff
@@ -6,24 +6,80 @@
+import SwiftUI
 
-enum MSError: Error {
+enum MSError: Int {
+    
+    /// Https
     case badUrl
     case badHttpRequest
     case networkConnection
     
-    var localizedDescription: String {
+    /// Storage
+    case notSaved
+    case notFound
+    case badData
+    case badKey
+    
+    /// Authentication
+    case badCredentials
+    case userNotFound
+    
+    /// Codable
+    case encodingError
+    case decodingError
+
+    var friendlyMessage: String {
         switch self {
-            case .badUrl: "Invalid URL"
-            case .badHttpRequest: "Bad http request"
-            case .networkConnection: "Network connection"
+        case .badHttpRequest: "Hubo un problema con la solicitud. Por favor, inténtalo de nuevo."
+        case .networkConnection: "Por favor, verifica tu conexión a internet e inténtalo de nuevo."
+        case .badCredentials, .userNotFound:
+                "Las credenciales son incorrectas. Por favor, inténtalo de nuevo."
+        default: "Algo salió mal. Por favor, intenta más tarde."
         }
     }
     
-    var friendlyMessage: String {
+    var debugDescription: String {
         switch self {
-            case .networkConnection: "Algo salió mal, revise su conexión a internet e intente de nuevo"
-            default: "Algo salió mal, intente de nuevo mas tarde"
+        case .badUrl: "Invalid URL format"
+        case .badHttpRequest: "HTTP request failed due to malformed request"
+        case .networkConnection: "Network connection error"
+        case .notSaved: "Data could not be saved to storage"
+        case .notFound: "Data not found in storage"
+        case .badData: "Corrupt or malformed data"
+        case .badKey: "Invalid key for data retrieval"
+        case .badCredentials: "Incorrect user credentials provided"
+        case .userNotFound: "User not found in the database"
+        case .encodingError: "Encoding error ocurred"
+        case .decodingError: "Decoding error ocurred"
         }
     }
 }
+
+// MARK: - MSError helper
+
+final class MSErrorHelper {
+    private var referenceClass: AnyObject?
+    
+    private lazy var classDescription: String = {
+        guard let referenceClass else { return "No references" }
+        return String(describing: referenceClass)
+    }()
+    
+    /// Initializes an error handler with a reference to the calling object for better error tracking.
+    ///
+    /// - Parameter referenceClass: A reference to the calling object, typically used to provide context
+    /// for logging and debugging where the error originated from in the system.
+    init(referenceClass: AnyObject?) {
+        self.referenceClass = referenceClass
+    }
+    
+    deinit { referenceClass = nil }
+    
+    // MARK: - Handler
+    
+    func handle(error: MSError) -> Error {
+        debugPrint("❌❌❌ Debug: \(error.debugDescription). Caller: \(classDescription) ❌❌❌")
+        return NSError(domain: error.friendlyMessage, code: error.rawValue)
```

Comment:
this code present  1, 2, 3 etc depends of the value, is this correct?

Reply:
I do not undertand your comment. Do you have any suggestions or is only a question?

Errors are interna managed by us. We can define values to each one.

Do you consider neccessary?